sdk/src/Services/S3/Custom/Util/AmazonS3Uri.cs

```diff
@@ -221,16 +226,14 @@ public static bool IsAmazonS3Endpoint(Uri uri)
             if (uri == null)
                 throw new ArgumentNullException("uri");
 
-            var match = new Regex(EndpointPattern).Match(uri.Host);
             if (uri.Host.EndsWith("amazonaws.com", StringComparison.OrdinalIgnoreCase) || uri.Host.EndsWith("amazonaws.com.cn", StringComparison.OrdinalIgnoreCase))
             {
-                return match.Success;
+                return EndpointRegexMatch.Match(uri.Host).Success;
```

Comment:
This could also be written as
```
EndpointRegexMatch.IsMatch(uri.Host);
```

Without knowing the performance differences, it seems [`IsMatch()`](https://source.dot.net/#System.Text.RegularExpressions/System/Text/RegularExpressions/Regex.Match.cs,5a9d5ededeb92966,references) takes some `quick` path where `.Match()` does not.

Reply:
Thanks for the tip.  This PR has already merged and shipped, but I'll save your suggestion next time I'm working with Regular Expressions.